Initium-Odp/src/com/universeprojects/miniup/server/commands/CommandSetLabel.java

```diff
@@ -0,0 +1,91 @@
+package com.universeprojects.miniup.server.commands;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.universeprojects.cacheddatastore.CachedDatastoreService;
+import com.universeprojects.cacheddatastore.CachedEntity;
+import com.google.appengine.api.datastore.Key;
+import com.universeprojects.miniup.server.ODPDBAccess;
+import com.universeprojects.miniup.server.WebUtils;
+import com.universeprojects.miniup.server.commands.framework.Command;
+import com.universeprojects.miniup.server.commands.framework.UserErrorMessage;
+
+
+/**
+ * Allows the current leader of a party to set a different party member as leader
+ * 
+ * Usage notes:
+ * Checks if itemId is a valid storage option, and in inventory/at the current location
+ * Also sanitises label for any possible injection ideas. 
+ * 
+ * Parameters:
+ * 		itemId - itemID of the chest to be relabelled 
+ * 		label - the new label
+ * 
+ * @author NJ
+ *
+ */
+public class CommandSetLabel extends Command {
+
+	public CommandSetLabel(HttpServletRequest request, HttpServletResponse response) 
+	{
+		super(request, response);
+	}
+
+	@Override
+	public void run(Map<String, String> parameters) throws UserErrorMessage 
+	{
+		// Get the DB access object. We use this for most DB communication. It is a place for reusable methods that access the DB.
+		ODPDBAccess db = getDB();
+
+		// Check if we're logged in or not. We shouldn't be able to see this since commands cannot be used unless the user is logged in.
+		if (db.isLoggedIn(request)==false)
```

Comment:
I THINK the isLoggedIn() call is not necessary because the Command class always checks if logged in and throws if not. Please test to make sure this assumption is correct before taking any action tho.